src/gui/properties/speedplotview.cpp

```diff
@@ -169,6 +171,67 @@ int SpeedPlotView::maxYValue()
     return maxYValue;
 }
 
+namespace
+{
+    // table of supposed nice steps for grid marks to get nice looking quarters of scale
+    const double roundingTable[] = {1.2, 1.6, 2, 2.4, 2.8, 3.2, 4, 6, 8, 10};
+
+    qint64 getNiceYScale(double value)
+    {
+        if (value == 0.0) return 0;
+        if (value <= 12.0) return 12;
+
+        qint64 scale = 1;
+        while (value > 1024) {
+            value /= 1024;
+            scale *= 1024;
+        }
+
+        if (value > 100.0) {
+            int roundedValue = static_cast<int>(value / 40) * 40;
+            while (roundedValue < value)
+                roundedValue += 40;
+            return scale * roundedValue;
+        }
+
+        if (value > 10.0) {
+            int roundedValue = static_cast<int>(value / 4) * 4;
+            while (roundedValue < value)
+                roundedValue += 4;
+            return scale * roundedValue;
+        }
+
+        for (const auto &roundedValue : roundingTable)
+            if (value <= roundedValue)
+                return static_cast<qint64>(roundedValue * scale);
+
+        return 10 * scale; // really we should not get here, but compiler complains
+    }
+
+    using Utils::Misc::SizeUnit;
+
+    QString niceLabel(qint64 bytesValue, SizeUnit &unit, bool fixed)
+    {
+        qreal friendlyVal;
+        if (fixed) {
+            // calculate friendlyVal from bytesVal using fixed unit size
+            friendlyVal = qreal(bytesValue) / Utils::Misc::sizeInBytes(1, unit);
+        }
+        else {
+            // calculate friendlyVal and non-fixed unit from bytesVal
+            if (!Utils::Misc::friendlyUnit(bytesValue, friendlyVal, unit))
+                return "Unknown";
+        }
+        if (unit == SizeUnit::Byte)
+            return QString::number(bytesValue)
```

Comment:
Please describe why these two can't be defined using "Utils::Misc::friendlyUnit()".

Reply:
>what did you mean by the words these two ?

I mean these two return statements.

Reply:
`niceLabel()` has a bit confusing definition and parameters. What is it supposed to be? Why "unit" is non-const reference (aka output parameter)?

Reply:
Well, may be some refactoring is already needed here. I'll try to do so. Marking as WIP meanwhile

Reply:
Current code uses friendlyUnit() for creating graph labels. IIRC, you stop to use it since it uses predefined precision values. Why not refactor it to accept precision as param?